source/simulator.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include "device.hpp"
+
+namespace sim {
+
+class Link;
+class Scheduler;
+class Flow;
+
+class Simulator {
+public:
+    Simulator();
+    void add_device(std::string a_name, DeviceType a_type);
+    void add_link(Device* a_node1, Device* a_node2, int a_delay);
```

Comment:
I think we can do naming better:
```suggestion
    void add_link(Device* a_node_from, Device* a_node_dest, int a_delay);
```